beacon-chain/builder/service.go

```diff
@@ -0,0 +1,52 @@
+package builder
+
+import (
+	"context"
+
+	types "github.com/prysmaticlabs/prysm/consensus-types/primitives"
+	v1 "github.com/prysmaticlabs/prysm/proto/engine/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+)
+
+type BlockBuilder interface {
+	SubmitBlindedBlock(ctx context.Context, block *ethpb.SignedBlindedBeaconBlockBellatrix) (*v1.ExecutionPayload, error)
+	GetHeader(ctx context.Context, slot types.Slot, parentHash [32]byte, pubKey [48]byte) (*ethpb.SignedBuilderBid, error)
+	Status() error
+	RegisterValidator(ctx context.Context, regs []*ethpb.SignedValidatorRegistrationV1) error
```

Comment:
I'd like to implement the client taking a single SignedValidatorRegistrationV1 rather than a slice. The REST API doesn't have a batch mode, so the client would be making multiple requests anyway. Error handling would be cleaner for the client and the consumer of this method with individual calls, vs trying to figure out a way to express partial success/failure of the batch.